nomos-sdp/src/lib.rs

```diff
@@ -211,3 +211,9 @@ impl<Metadata, Proof> SdpMessage<Metadata, Proof> {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct FinalizedBlockEvent {
+    pub block_number: BlockNumber,
+    pub updates: Vec<(ProviderInfo, DeclarationUpdate)>,
```

Comment:
After giving it another though, it seems that it would be better to define a `ProviderInfoUpdate`, since we Membership service doesn't have pruning yet, it's not that obvious, but ProviderInfo in itself doesn't have the state, because the state depends on the block number, in sdp ledger it's derived by converting ProviderInfo with 

``` 
impl ProviderState {
    pub fn try_from_info (
```

in `nomos-sdp/src/state.rs`, but here we could return `ProviderInfoUpdate` defined something like this in `nomos-sdp/src/lib.rs`:

```
enum ProviderState {
  Active, Inactive, Withdrawn
}

struct ProviderUpdate {
  provider_id: ProviderId,
  state: ProviderState,
}
```

Reply:
Not sure if `Event` definition from the spec could be mapped to suggested `ProviderUpdate`. For this PR, lets just define and return what is needed for the Membership service - `ProviderId` + `State` + `BlockNumber` + `Locators`.

Reply:
Sorry, when I said `Event` I met this event `FinalizedBlockEvent`. 

Ok, sounds good.